app/subcommands/master_subcommand.py

```diff
@@ -43,4 +43,6 @@ def async_run(self, port, log_level, eventlog_file):
 
         ioloop.start()  # this call blocks until the server is stopped
         ioloop.close(all_fds=True)  # all_fds=True is necessary here to make sure connections don't hang
+        self._logger.notice('About to shut down master service, saving all active builds to database.')
+        cluster_master._build_store.clean_up()  # pylint: disable=protected-access
```

Comment:
This might not be in the right spot -- is there a better place to call this method when the master shuts down?